src/Reporter.cpp

```diff
@@ -366,7 +366,7 @@ namespace geopm
 
         comm->gatherv((void *) (host_report.c_str()), sizeof(char) * buffer_size,
                       (void *) report_buffer.data(), buffer_size_array, buffer_displacement, 0);
-        report_buffer.back() = '\0';
+        report_buffer.push_back('\0');
```

Comment:
This *could* require the creation of a copy of the report.  Probably should just set the last character to null conditioned on 
```
if (report_buffer.size() != 0) {
    report_buffer.back() = '\0';
}
```